### DISCUSSION TITLE
SMILES to ZINC Data in torch_geometric conversion function

### DISCUSSION THREAD

**Discussion by andreibrasoveanu97** [pyg-team/pytorch_geometric#9419] [Q&A]
Hello,

I've trained some models on the ZINC dataset and I would like to use them to predict the values for molecules outside the dataset, given under a SMILES format. I've seen [this discussion](https://github.com/pyg-team/pytorch_geometric/discussions/3451) on this forum but I've spent hours reading through the links suggested there with no result. 

I've found plenty of methods of converting SMILES to `torch_geometric.Data` objects, but not to the same format that is found within the `torch_geometric.datasets.ZINC` objects. 

Does anyone know where could I find the function or a mapping of some kind to get the same result? 

Thanks!

**Comment by rusty1s** [comment]
You can find the pre-processing code of `ZINC` [here](https://github.com/graphdeeplearning/benchmarking-gnns/tree/master/data/molecules).

**Reply by andreibrasoveanu97** [reply]
This does not show however how are the molecules (or graphs) built, I am looking for the parsing method of the `SMILES` of a molecule to the actual graph (or `Data` object).

**Reply by rusty1s** [reply]
In this case I suggest to open an issue at https://github.com/graphdeeplearning/benchmarking-gnns
